fsspec/implementations/zip.py

```diff
@@ -132,3 +132,59 @@ def _open(
             out.size = info["size"]
             out.name = info["name"]
         return out
+
+    def find(self, path, maxdepth=None, withdirs=False, detail=False, **kwargs):
+        if maxdepth is not None and maxdepth < 1:
+            raise ValueError("maxdepth must be at least 1")
+
+        result = {}
+
+        def _below_max_recursion_depth(path):
```

Comment:
maxdepth really exists for filesystems doing an explicit ls(), to prevent having to walk very deep trees. In this case, we consider all files anyway, so the code can be much simpler by ignoring maxdepth.

Reply:
I agree that removing it would make this quite a lot simpler. I wondering, since `maxdepth` is part of the interface if you think that there is any risk that people using it are relying on it only returning files from a specific `maxdepth`? I think that might be a case for keeping it in?